src/main/java/com/blamejared/crafttweaker/impl/helper/ItemStackHelper.java

```diff
@@ -0,0 +1,63 @@
+package com.blamejared.crafttweaker.impl.helper;
+
+import com.blamejared.crafttweaker.api.data.NBTConverter;
+import com.blamejared.crafttweaker.impl.data.MapData;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+
+public final class ItemStackHelper {
+    
+    public static String getCommandString(final ItemStack stack) {
+    
+        final StringBuilder sb = new StringBuilder("<item:");
+        sb.append(stack.getItem().getRegistryName());
+        sb.append('>');
+    
+        if(stack.getTag() != null) {
+            MapData data = (MapData) NBTConverter.convert(stack.getTag()).copyInternal();
+            //Damage is special case, if we have more special cases we can handle them here.
+            if(stack.getItem().isDamageable()) {
+                data.remove("Damage");
+            }
+            if(!data.isEmpty()) {
+                sb.append(".withTag(");
+                sb.append(data.asString());
+                sb.append(')');
+            }
+        }
+    
+        if(stack.getDamage() > 0) {
+            sb.append(".withDamage(").append(stack.getDamage()).append(')');
+        }
+    
+        if(!stack.isEmpty()) {
+            if(stack.getCount() != 1) {
+                sb.append(" * ").append(stack.getCount());
+            }
+        }
+        return sb.toString();
+    }
+    
+    // TODO("This is a copy of IItemStack#matches written to avoid object creation: find a way to avoid code duplication")
+    public static boolean areStacksTheSame(final ItemStack first, final ItemStack second) {
```

Comment:
It can't delegate here if this check is removed
https://github.com/CraftTweaker/CraftTweaker/pull/1367/files#diff-284fda9bdb66c62b5c5a95511858493691c2f14940a29cb70a465f5ee15a63ceR46
unless another param was added like `final boolean checkSize`?